scalyr_agent/agent_main.py

```diff
@@ -1658,6 +1749,59 @@ def __calculate_overall_stats(self, base_overall_stats):
             result.rss_size,
         ) = self.__controller.get_usage_info()
 
+        if copy_manager_warnings:
+            result.avg_bytes_copied_rate = (
+                result.total_bytes_copied - self.__last_total_bytes_copied
+            ) / self.__config.copying_manager_stats_log_interval
+            result.avg_bytes_produced_rate = (
```

Comment:
You could define two variables here called `total_bytes_produced` and `last_total_bytes_produced`.  It'd probably make the code a little more readable.  Alternatively, add a comment here that says you compute total produced by adding bytes_skipped, bytes_copied, and bytes_pending.